### PR TITLE
Remove ngraph options

### PR DESCRIPTION
**Pull request by ilyachur**
### Details:
 - Renamed NGRAPH_ cmake options
 - Renamed `unit-test` to `ov_core_unit_tests`

### Tickets:
 - *ticket-id*

### CODE REVIEW

File: .ci/azure/linux.yml [hunk 1]
```diff
@@ -152,7 +152,7 @@ jobs:
         -DPYTHON_EXECUTABLE=/usr/bin/python3.8
         -DENABLE_WHEEL=ON
         -DENABLE_TESTS=ON
-        -DNGRAPH_ONNX_FRONTEND_ENABLE=ON
+        -DOV_ONNX_FRONTEND_ENABLE=ON
```
Comment (ilya-lavrenov) [inline .ci/azure/linux.yml:155]: it does not match all other options in cmake. We have rule like `ENABLE_xxx`

File: cmake/features.cmake [hunk 2]
```diff
@@ -160,33 +160,33 @@ else()
...
-ie_dependent_option(NGRAPH_UNIT_TEST_ENABLE "Enables ngraph unit tests" ON "ENABLE_TESTS;NOT ANDROID" OFF)
-ie_dependent_option(NGRAPH_UNIT_TEST_BACKENDS_ENABLE "Control the building of unit tests using backends" ON
-    "NGRAPH_UNIT_TEST_ENABLE" OFF)
-ie_option(OPENVINO_DEBUG_ENABLE "Enable output for OPENVINO_DEBUG statements" OFF)
+ie_dependent_option(ENABLE_OV_ONNX_FRONTEND "Enable ONNX FrontEnd" ON "protoc_available" OFF)
+ie_dependent_option(ENABLE_OV_PDPD_FRONTEND "Enable PaddlePaddle FrontEnd" ON "protoc_available" OFF)
+ie_option(OV_IR_FRONTEND_ENABLE "Enable IR FrontEnd" ON)
+ie_dependent_option(ENABLE_OV_TF_FRONTEND "Enable TensorFlow FrontEnd" ON "protoc_available" OFF)
+ie_dependent_option(OV_USE_SYSTEM_PROTOBUF "Use system protobuf" OFF
+    "ENABLE_OV_ONNX_FRONTEND OR ENABLE_OV_PDPD_FRONTEND OR ENABLE_OV_TF_FRONTEND;BUILD_SHARED_LIBS" OFF)
```
Comment (ilya-lavrenov) [inline cmake/features.cmake:168]: why `OV_USE_SYSTEM_PROTOBUF` and `OV_IR_FRONTEND_ENABLE ` are not aligned?

Comment (ilyachur) [reply cmake/features.cmake:168]: Fixed

File: cmake/features.cmake [hunk 3]
```diff
@@ -160,33 +160,33 @@ else()
...
+ie_dependent_option(ENABLE_OV_ONNX_FRONTEND "Enable ONNX FrontEnd" ON "protoc_available" OFF)
+ie_dependent_option(ENABLE_OV_PDPD_FRONTEND "Enable PaddlePaddle FrontEnd" ON "protoc_available" OFF)
+ie_option(ENABLE_OV_IR_FRONTEND "Enable IR FrontEnd" ON)
+ie_dependent_option(ENABLE_OV_TF_FRONTEND "Enable TensorFlow FrontEnd" ON "protoc_available" OFF)
+ie_dependent_option(ENABLE_SYSTEM_PROTOBUF "Use system protobuf" OFF
+    "ENABLE_OV_ONNX_FRONTEND OR ENABLE_OV_PDPD_FRONTEND OR ENABLE_OV_TF_FRONTEND;BUILD_SHARED_LIBS" OFF)
+ie_dependent_option(ENABLE_OV_CORE_UNIT_TESTS "Enables OpenVINO core unit tests" ON "ENABLE_TESTS;NOT ANDROID" OFF)
+ie_dependent_option(ENABLE_OV_CORE_BACKEND_UNIT_TESTS "Control the building of unit tests using backends" ON
+    "ENABLE_OV_CORE_UNIT_TESTS" OFF)
+ie_option(ENABLE_OPENVINO_DEBUG "Enable output for OPENVINO_DEBUG statements" OFF)
```
Comment (ilya-lavrenov) [inline cmake/features.cmake:172]: if we just rename option, some private CI maybe incorrect. E.g. private CI disables ONNX FE, after such renaming this `NGRAPH_TF_FRONTEND_ENABLE=OFF` option does nothing.

Can we have aliases for keys like `ie_dependent_option("ENABLE_OV_CORE_BACKEND_UNIT_TESTS;NGRAPH_UNIT_TEST_BACKENDS_ENABLE " "Control the building of unit tests using backends" ON
    "ENABLE_OV_CORE_UNIT_TESTS" OFF)`
Where both keys are valid and synchronized

See how it's done for GPU https://github.com/openvinotoolkit/openvino/blob/88e078460629bc616e5b52b2f03df375e6f52a7b/cmake/features.cmake#L15-L16

Comment (ilya-lavrenov) [reply cmake/features.cmake:172]: CC @vinograd47 do you have advices how to achieve this?

**Review by ilya-lavrenov — APPROVED** [review]

**Review by ilya-lavrenov — APPROVED** [review]

**Comment by azhogov** [comment]
CI passed: https://github.com/intel-innersource/frameworks.ai.openvino.ci.product-configs/pull/634

**Pull request merged**